shoop/admin/modules/orders/views/create.py

```diff
@@ -86,28 +144,84 @@ def handle_product_data(self, request):
         product_id = request.GET["id"]
         shop_id = request.GET["shop_id"]
         customer_id = request.GET.get("customer_id")
+        quantity = decimal.Decimal(request.GET.get("quantity", 1))
         product = Product.objects.get(pk=product_id)
+        if not product:
+            return {"errorText": _("Product %s does not exist.") % product_id}
         shop = Shop.objects.get(pk=shop_id)
-        ctx_request = RequestFactory().get("/")
-        ctx_request.shop = shop
-        ctx_request.customer = Contact.objects.filter(pk=customer_id).first()
-        ctx_request.user = AnonymousUser()
-        context = get_pricing_module().get_context_from_request(ctx_request)
-        price_info = product.get_price_info(context, quantity=1)
+        try:
+            shop_product = product.get_shop_instance(shop)
+        except ObjectDoesNotExist:
+            return {"errorText": _("Product %(product)s is not available in the %(shop)s shop.") % {
+                "product": product.name, "shop": shop.name}}
+
+        min_quantity = shop_product.minimum_purchase_quantity
+        # Make quantity to be at least minimum quantity
+        quantity = min_quantity if quantity < min_quantity else quantity
+        customer = Contact.objects.filter(pk=customer_id).first() if customer_id else None
+        price_info = get_price_info(shop, customer, product, quantity)
+        error_text = None
+        supplier = shop_product.suppliers.first()  # TODO: Allow setting a supplier?
+        for message in shop_product.get_orderability_errors(
+                supplier=supplier, quantity=quantity, customer=customer, ignore_minimum=True):
+            error_text = str(message.args[0])
```

Comment:
Use something like [is already done in the product edit view](https://github.com/shoopio/shoop/blob/5a68a8a1ae8c42cfb9ec18912af1f5963bd97a3f/shoop/admin/modules/products/views/edit.py#L173-L183) to get _all_ orderability errors, not just the last one.